stan/math/opencl/matrix_cl.hpp

```diff
@@ -9,7 +9,8 @@
 #include <stan/math/opencl/err/check_opencl.hpp>
 #include <stan/math/prim/arr/fun/vec_concat.hpp>
 #include <stan/math/prim/scal/err/check_size_match.hpp>
-#include <cl.hpp>
+#include <stan/math/prim/scal/err/domain_error.hpp>
```

Comment:
Why do we need this header here? I see no domain_error (deprecated since [yesterday](https://github.com/stan-dev/math/pull/1482)) or throw_domain_error here?

Reply:
Oops, good catch. I wasn't paying attention when rebasing onto HEAD.